### PR TITLE
Implemented env dependent pushing of information

### PR DESCRIPTION
**Pull request by gentrexha**
`environment` now is a text instead of a choice variable which allows environment dependent pushing of information.

### CODE REVIEW

File: dune_api_scripts/update_appdata_view.py [hunk 1]
```diff
@@ -20,18 +20,17 @@ def refresh(dune: DuneAPI, query: DuneQuery):
...
 
-class Environment(Enum):
-    """Enum for Deployment Environments"""
-
-    STAGING = "barn"
-    PRODUCTION = "prod"
+@dataclass
+class Environment:
+    """Dataclass for Deployment Environments"""
+    value: str
```
Comment (bh2smith) [inline dune_api_scripts/update_appdata_view.py:26]: I was kindof hoping we could just add a third enum value like `TEST = "test"` which is the default (when none is provided as a script argument). 

This gave us some security when it comes to not arbitrarily / accidentally creating several different variations of the same table. 

If some one runs the script now with random input data we will find that several different copies of this able will be created in Dune.

**Review by bh2smith — COMMENTED** [review]
While I think this is an approach that will work, I am not so sure its great to allow any environment string the user chooses when it come to creating table names.

Screenshot 2022-07-18 at 1 02 37 PM

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ data/
 .vscode
 env/
 venv/
+.idea/
```
[the remark this answers is not in the document]
Comment (gentrexha) [reply .gitignore:9]: Haven't done this before actually. I like how you're keeping the repository editor agnostic. Might wanna remove `.vscode` too then.

File: dune_api_scripts/update_appdata_view.py [hunk 3]
```diff
@@ -45,7 +44,7 @@ def update_raw_app_data(dune: DuneAPI, env: Environment):
         ),
         network=Network.MAINNET,
         parameters=[env.as_query_param()],
-        query_id=int(getenv("QUERY_ID_RAW_APP_DATA", "1032460")),
+        query_id=int(getenv("QUERY_ID_RAW_APP_DATA", "1044750")),
```
[the remark this answers is not in the document]
Comment (gentrexha) [reply dune_api_scripts/update_appdata_view.py]: I totally missed this. Thanks for pointing that out.

File: dune_api_scripts/update_appdata_view.py (hunk 1 above)
Comment (gentrexha) [reply dune_api_scripts/update_appdata_view.py]: Makes sense to not allow free text creation of views in Dune. I've replaced the current implementation with the `test` env variable you suggested.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (bh2smith) [reply .gitignore]: Yes definitely lets remove that too. Could you remove it here?

File: dune_api_scripts/update_appdata_view.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bh2smith) [reply dune_api_scripts/update_appdata_view.py]: I guess its not super obvious, but now you know!

File: dune_api_scripts/update_appdata_view.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 from duneapi.types import DuneQuery, Network, QueryParameter
 from duneapi.util import open_query
 
-from .utils import app_data_entries
+from dune_api_scripts.utils import app_data_entries
```
Comment (bh2smith) [inline dune_api_scripts/update_appdata_view.py:10]: Nice! Good catch here.

File: dune_api_scripts/update_appdata_view.py [hunk 5]
```diff
@@ -25,13 +25,14 @@ class Environment(Enum):
...
     PRODUCTION = "prod"
+    TEST = "test"
 
     def __str__(self) -> str:
         return self.value
 
     def as_query_param(self) -> QueryParameter:
         """Converts Environment to Dune Query Parameter"""
-        return QueryParameter.enum_type("Environment", self.value, ["barn", "prod"])
+        return QueryParameter.enum_type("Environment", self.value, ["barn", "prod", "test"])
```
Comment (bh2smith) [inline dune_api_scripts/update_appdata_view.py:35]: I wonder if there is some way for the Enum class to generate this list. Best I found was this:

```
[e.value for e in Environment]
```

What do you think about including that here. Then future devs won't have to update two independent places.

**Review by bh2smith — APPROVED** [review]
Looks good, maybe lets just use a generated list of Enum Parameters.

File: dune_api_scripts/update_appdata_view.py (hunk 4 above)
Comment (gentrexha) [reply dune_api_scripts/update_appdata_view.py:10]: That was actually supposed to be a part of the suggestions PR, but thanks!

**Comment by bh2smith** [comment]
I think the only thing missing now is that "test" is the default argument chosen when nothing is passed into the script arguments.
